code/cynthia/Python/test_ATM_lab12.py

```diff
@@ -0,0 +1,37 @@
+#Automated Tests
+#using using ATM 
+
+from lab08_2 import ATM
+import pytest
+from unittest.mock import patch 
+
+def test_ATM_creation():
+    atm = ATM(0)
+    assert atm.balance == 0
+    atm = ATM(-5)
+    assert atm.balance == -5
+    atm = ATM(.1)
+    assert atm.interest_rate == .1
+
+def test_check_balance():
+    atm = ATM(0)
+    assert atm.check_balance() == 0
+
+def test_deposit():
+    atm = ATM(50)
+    assert atm.deposit(10) == 60
+    assert atm.deposit(30)  == 90
+
+def test_check_withdrawal():
+    atm = ATM(10)
+    assert atm.check_withdrawal(5) == True
+   
+
+def test_withdraw():
+    atm = ATM(50)
+    assert atm.withdraw(10) == 40
+    assert atm.withdraw(5) == 35
+
+def test_calc_interest():
+    atm = ATM(50)
+    assert atm.interest_rate == .1
```

Comment:
You haven't actually called the `calc_interest` method here, you're only checking what the interest rate is